unicon_backend/workers/consumer.py

```diff
@@ -27,13 +36,53 @@ def message_callback(
         response: JobResult = JobResult.model_validate_json(body)
         with SessionLocal() as db_session:
             task_result = db_session.scalar(
-                select(TaskResultORM).where(TaskResultORM.job_id == str(response.id))
+                sa.select(TaskResultORM).where(col(TaskResultORM.job_id) == response.id)
             )
             if task_result is not None:
                 task_result.status = TaskEvalStatus.SUCCESS
-                task_result.result = response.model_dump(mode="json")  # TEMP
                 task_result.completed_at = sa.func.now()  # type: ignore
 
+                # Evaluate result based on OutputStep
+                task = cast(ProgrammingTask, task_result.task_attempt.task.to_task())
+
+                # Assumption: testcase index = result index
+                # Updates to testcases must be done very carefully because of this assumption.
+                # To remove this assumption, we need to add a testcase_id field to the result model.
```

Comment:
I believe `testcase_id` is already part of the tracking fields? it should be part of `ProgramResult` sent back from the runner:

https://github.com/uniconhq/unicon-backend/blob/3f36d7bd79233fb93ed57168cf7cd46c00f5607c/unicon_backend/runner.py#L35-L41

Reply:
hmm is this a old comment? the implementation below makes use of the tracking field